src/tir/analysis/deep_equal.cc

```diff
@@ -62,7 +67,7 @@ bool ExprDeepEqual::operator()(const PrimExpr& lhs, const PrimExpr& rhs) const {
   if (lhs.as<AnyNode>()) {
     return false;
   }
-  return DeepCmpSEqualHandler().SEqualReduce(lhs, rhs, false);
+  return DeepCmpSEqualHandler().SEqualReduce(lhs, rhs, false, {});
```

Comment:
nit: prefer more explicit instantiation of `Optional<>`

```suggestion
  return DeepCmpSEqualHandler().SEqualReduce(lhs, rhs, false, NullOpt);
```